TAGradingServer/account/ajax/account-numerictext-gradeable.php

```diff
@@ -61,33 +76,41 @@
-    if (isset($tmp["gcd_score"])){
-        //Numeric item - update score
-        if ($i<=$num_numeric){
-            $params = array($questions[$i-1],$tmp['gc_id'],$tmp['gd_id']);
-            $db->query('UPDATE gradeable_component_data SET gcd_score=? WHERE gc_id=? AND gd_id=?', $params);
-        }else{ // text item update comment
-            $params = array($text[$i-$num_numeric-1], $tmp['gc_id'],$tmp['gd_id']);
-            $db->query('UPDATE gradeable_component_data SET gcd_component_comment=? WHERE gc_id=? AND gd_id=?', $params);
-        }
-    } 
-    else{
-        // CREATE THE new score
-        //Numeric item - update score
-        //find the gradeable component id
-        $params = array($g_id, $i);
-        $db->query("SELECT gc_id FROM gradeable AS g INNER JOIN gradeable_component AS gc ON g.g_id=gc.g_id WHERE g.g_id=? AND gc.gc_order=?",$params);
-        $gc_id = $db->row()['gc_id'];
-        if ($i<=$num_numeric){
-            $params = array($gc_id,$gd_id, $questions[$i-1],'');
-        }else{ // text item update comment
-            $params = array($gc_id,$gd_id, 0,$text[$i-$num_numeric-1]);
+        $tmp = $db->row();
+
+        //UPDATE the existing score
+        if (isset($tmp["gcd_score"])) {
+            //Numeric item - update score
+            if ($i <= $num_numeric) {
+                $params = array($questions[$i - 1], $tmp['gc_id'], $tmp['gd_id']);
+                $db->query('UPDATE gradeable_component_data SET gcd_score=? WHERE gc_id=? AND gd_id=?', $params);
+            } else { // text item update comment
+                $params = array($text[$i - $num_numeric - 1], $tmp['gc_id'], $tmp['gd_id']);
+                $db->query('UPDATE gradeable_component_data SET gcd_component_comment=? WHERE gc_id=? AND gd_id=?', $params);
+            }
+        } else {
+            // CREATE THE new score
+            //Numeric item - update score
+            //find the gradeable component id
+            $params = array($g_id, $i);
+            $db->query("SELECT gc_id FROM gradeable AS g INNER JOIN gradeable_component AS gc ON g.g_id=gc.g_id WHERE g.g_id=? AND gc.gc_order=?", $params);
+            $gc_id = $db->row()['gc_id'];
+            if ($i <= $num_numeric) {
+                $params = array($gc_id, $gd_id, $questions[$i - 1], '');
+            } else { // text item update comment
+                $params = array($gc_id, $gd_id, 0, $text[$i - $num_numeric - 1]);
+            }
+            $db->query("INSERT INTO gradeable_component_data (gc_id, gd_id, gcd_score, gcd_component_comment) VALUES (?,?,?,?)", $params);
         }
-        $db->query("INSERT INTO gradeable_component_data (gc_id, gd_id, gcd_score, gcd_component_comment) VALUES (?,?,?,?)",$params);
     }
 }
 
+function csvupload(){
+
+    $csv = $_POST['parsedCsv'];
+    global $db;
+    $params = array('{"'.implode('","', $csv).'"}', $_GET["g_id"], User::$user_id);
+
+    $db->query("SELECT * FROM csv_To_Numeric_Gradeable(?, ?, ?)", $params);
```

Comment:
Wouldn't this make more sense to do `SELECT csv_To_Numeric(?, ?, ?)`?

And what's the benefit of doing all of this stuff directly in the database as a function than using PHP & DB to handle this stuff? Using purely DB does make it a tad harder to easily update it if we decide to change some part of the code later.